### PR TITLE
ft: implement user logout feature

### LINKED ISSUE #43: Feat: User Logout
### Description

As a user (vendor or buyer) of the e-commerce platform's backend API, I want to be able to log out securely to terminate my current session and prevent unauthorized access to my account.

### Acceptance Criteria

- [ ] Implement a user logout API endpoint
- [ ] Invalidate or revoke the user's current session or access token
- [ ] Clear any sensitive data stored server-side associated with the user's session
- [ ] Return appropriate success or error responses

### Developer Notes

- Identify the method used for user session management (e.g., session cookies, JSON Web Tokens)
- Implement the necessary logic to invalidate or revoke the user's current session or access token
- Clear any sensitive user data stored server-side associated with the session
- Handle edge cases, such as concurrent requests or expired sessions
- Write unit tests to cover different scenarios and edge cases
- Write clear API documentation for the resulting endpoint

### Related Issues

- #40 Feat: User Sign In with Email and Password 

### Definition of Done

- User logout API endpoint is implemented and functional
- User's current session or access token is invalidated or revoked
- Sensitive server-side data associated with the user's session is cleared
- Appropriate success and error responses are returned
- Unit tests are written and passing
- Code adheres to coding standards and best practices
- API documentation for the resulting endpoint is completed

### PR DESCRIPTION
**Pull request by Ndevu12**
#### What does this PR do?

This pull request aims to add feature that enables a user to logout

#### Description of Task to be completed?

The task completed in this pull request includes:

- check if a user has access or logged in
- access and clear user's access token from cookie
- provide success message for success and error message else

#### How should this be manually tested?

To manually test all the changes made in this pull request, follow these steps:

```
git clone -b ft-logout https://github.com/atlp-rwanda/knights-ecomm-be.git
cd knights-ecomm-be
npm i
npm run dev
```

### CODE REVIEW

**Review by maxCastro1 — APPROVED** [review]

File: src/services/userServices/userResendOTP.ts [hunk 1]
```diff
@@ -13,18 +13,22 @@ export const userResendOtpService = async (req: Request, res: Response) => {
...
+    console.log('User not found');
     return res.status(404).json({ status: 'error', message: 'Incorrect email' });
   }
 
   const otpCode = await start2FAProcess(user.email);
+  if (!otpCode) throw new Error('Error generating OTP');
   const OTPEmailcontent = otpTemplate(user.firstName, otpCode.toString());
+  if (!OTPEmailcontent) throw new Error('Error generating OTP email content');
   await sendOTPEmail('Login OTP', user.email, OTPEmailcontent);
   if (process.env.APP_ENV !== 'test') {
```
Comment (elijahladdie) [inline src/services/userServices/userResendOTP.ts:33]: Remove consoles please

File: src/services/userServices/userSendOTPEmail.ts [hunk 2]
```diff
@@ -21,7 +21,6 @@ export const sendOTPEmail = async (subject: string, email: string, content: any)
 
   try {
     const info = await transporter.sendMail(mailOptions);
-    console.log('Message sent: %s', info.messageId);
   } catch (error) {
     console.log('Error occurred while sending email', error);
   }
```
Comment (elijahladdie) [inline src/services/userServices/userSendOTPEmail.ts:26]: remove console.log here as well

**Review by elijahladdie — APPROVED** [review]

**Review by faid-terence — APPROVED** [review]
Nice Work @Ndevu12 !
